app/controllers/customers_controller.rb

```diff
@@ -0,0 +1,97 @@
+class CustomersController < ApplicationController
+  rescue_from ActiveRecord::RecordNotFound, with: :catch_not_found
+  before_action :set_customer, only: %i[show edit update destroy]
+
+  # GET /customers or /customers.json
+  def index
+    @customers = Customer.all
+  end
+
+  # GET /customers/1 or /customers/1.json
+  def show
+  end
+
+  # GET /customers/new
+  def new
+    @customer = Customer.new
+  end
+
+  # GET /customers/1/edit
+  def edit
+  end
+
+  # POST /customers or /customers.json
+  def create
+    @customer = Customer.new(customer_params)
+    if @customer.save
+      flash.notice = 'The customer record was created successfully.'
+      redirect_to @customer
+    else
+      render :new, status: :unprocessable_entity
+    end
+    # @customer = Customer.new(customer_params)
+
+    # respond_to do |format|
+    #   if @customer.save
+    #     format.html { redirect_to customer_url(@customer), notice: "Customer was successfully created." }
+    #     format.json { render :show, status: :created, location: @customer }
+    #   else
+    #     format.html { render :new, status: :unprocessable_entity }
+    #     format.json { render json: @customer.errors, status: :unprocessable_entity }
+    # end
+    # end
+  end
+
+  # PATCH/PUT /customers/1 or /customers/1.json
+  def update
+    if @customer.update(customer_params)
+      flash.notice = 'The customer record was updated successfully.'
+      redirect_to @customer
+    else
+      render :edit, status: :unprocessable_entity
+    end
+    #   respond_to do |format|
+    #     if @customer.update(customer_params)
+    #       format.html { redirect_to customer_url(@customer), notice: "Customer was successfully updated." }
+    #       format.json { render :show, status: :ok, location: @customer }
+    #     else
+    #       format.html { render :edit, status: :unprocessable_entity }
+    #       format.json { render json: @customer.errors, status: :unprocessable_entity }
+    #     end
+    # end
+  end
+
+  # DELETE /customers/1 or /customers/1.json
+  def destroy
+    begin
+      @customer.destroy
+      flash.notice = 'The customer record was successfully deleted'
+    rescue ActiveRecord::InvalidForeignKey
+      flash.notice = 'That customer record could not be deleted, because the customer has orders'
+    end
```

Comment:
This is a fine choice for how to handle attempting to delete a customer! It's great that you're only rescuing `InvalidForeignKey` exceptions so that we don't show this message when the customer can't be destroyed for some other reason. 👍 (Though if we add customer_id as a foreign key to another table, we would need to update this message or maybe change the behavior. This is just extra, FYI -- no need to make any changes.)